packages/react-i18n/src/index.tsx

```diff
@@ -66,9 +58,29 @@ export const useI18n = (): I18nProps => {
  * }
  * export default withI18n( MyComponent );
  */
-export const withI18n = createHigherOrderComponent< I18nProps >( ( InnerComponent ) => {
+export const withI18n = createHigherOrderComponent< I18nReact >( ( InnerComponent ) => {
 	return ( props ) => {
 		const i18n = useI18n();
 		return <InnerComponent { ...i18n } { ...props } />;
 	};
 }, 'withI18n' );
+
+/**
+ * Utility to make a new context value
+ *
+ * @param localeData The localeData
+ *
+ * @returns The context value with bound translation functions
+ */
+function makeContextValue( localeData?: LocaleData ): I18nReact {
+	const i18n = createI18n( localeData );
+	const i18nLocale = localeData?.[ '' ]?.localeSlug ?? 'en';
+	return {
+		__: i18n.__.bind( i18n ),
+		_n: i18n._n.bind( i18n ),
+		_nx: i18n._nx.bind( i18n ),
+		_x: i18n._x.bind( i18n ),
+		isRTL: i18n.isRTL.bind( i18n ),
+		i18nLocale,
```

Comment:
This may be very WordPress.com specific. I had an implementation that doesn't provide `i18nLocale`, but this property is already used in Gutenboarding.